sphinx_needs/needs.py

```diff
@@ -159,6 +165,28 @@
 LOGGER = get_logger(__name__)
 
 
+def load_schemas_config_from_json(app: Sphinx, config: _SphinxConfig) -> None:
+    """Merge the configuration from the JSON file into the Sphinx config."""
+    needs_config = NeedsSphinxConfig(config)
+    if needs_config.schemas_from_json is None:
+        return
+    json_file = Path(app.confdir, needs_config.schemas_from_json).resolve()
+
+    if not json_file.exists():
+        raise NeedsConfigException(
+            f"'sn_schema_from_json' file does not exist: {json_file}"
+        )
+
+    try:
+        with Path(json_file).open("rb") as fp:
+            json_data = json.load(fp)
+        assert isinstance(json_data, dict), "Data must be a dict"
+    except Exception as exc:
+        raise NeedsConfigException(f"Could not load JSON file: {exc}") from exc
+
+    config["needs_schemas"] = json_data
```

Comment:
Please use `NeedsSphinxConfig` for accessing this key, so that type safety is enforced.
I guess we should also check that not both `needs_schemas` and `needs_schemas_from_json` are set at the same time, and warn about this, so there is no silent override.

Reply:
Done, however type checking happens later in detail, left a comment for that.
I added a check if both schema_definitions and schema_definitions_from_json are given.